firmware/ec/test/suites/Test_ocmp_ltc4015.c

```diff
@@ -0,0 +1,833 @@
+
+/* ============================= Boilerplate ================================ */
+void suite_setUp(void)
+{
+    FakeGpio_registerDevSimple(LTC4015_GpioPins, LTC4015_GpioConfig);
+    fake_I2C_init();
+
+    fake_I2C_registerDevSimple(gbc_pwr_int_bat_charger.cfg.i2c_dev.bus,
+                    gbc_pwr_int_bat_charger.cfg.i2c_dev.slave_addr,
+                LTC4015_regs,
+                               sizeof(LTC4015_regs), sizeof(LTC4015_regs[0]),
+                               sizeof(uint8_t), FAKE_I2C_DEV_LITTLE_ENDIAN);
+
+    fake_I2C_registerDevSimple(0, 0x45, SX1509_regs,
+                               sizeof(SX1509_regs), sizeof(SX1509_regs[0]),
+                               sizeof(uint8_t), FAKE_I2C_DEV_LITTLE_ENDIAN);
+}
+
+void setUp(void)
+{
+    memset(LTC4015_regs, 0, sizeof(LTC4015_regs));
+
+    OcGpio_init(&gbc_io_1);
+
+    LTC4015_init(&gbc_pwr_int_bat_charger);
+}
+
+void tearDown(void)
+{
+}
+
+void suite_tearDown(void)
+{
+    fake_I2C_deinit(); /* This will automatically unregister devices */
+}
+// Parameters are not used as this is just used to test assigning the
+//   alert_handler right now.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+void OCMP_GenerateAlert(const AlertData *alert_data,
+                        unsigned int alert_id,
+                        const void *data)
+{
+
+}
+#pragma GCC diagnostic pop
+/* ================================ Tests =================================== */
+void test_ocmp_ltc4015_probe(void)
+{
+    POSTData postData;
+    LTC4015_GpioConfig[0x05] = 0;
+
+    /* Test with the actual value */
+    LTC4015_regs[0x39] = LTC4015_CHARGER_ENABLED;
+    TEST_ASSERT_EQUAL(POST_DEV_FOUND, LTC4015_fxnTable.cb_probe(
+                &gbc_pwr_int_bat_charger, &postData));
+    TEST_ASSERT_EQUAL(OCGPIO_CFG_OUTPUT, LTC4015_GpioConfig[0x05]);
+
+     /* Test with an incorrect value */
+    LTC4015_regs[0x39] = ~LTC4015_CHARGER_ENABLED;
+    TEST_ASSERT_EQUAL(POST_DEV_MISSING, LTC4015_fxnTable.cb_probe(
+                &gbc_pwr_int_bat_charger, &postData));
+    /* Test with a missing device */
+    TEST_ASSERT_EQUAL(POST_DEV_MISSING, LTC4015_fxnTable.cb_probe(
+                            &gbc_pwr_invalid_dev, &postData));
+
+    /* Test with a missing bus */
+    TEST_ASSERT_EQUAL(POST_DEV_MISSING, LTC4015_fxnTable.cb_probe(
+                            &gbc_pwr_invalid_bus, &postData));
+}
+
+void test_ocmp_ltc4015_init(void)
+{
+    /* Now try to init with a pin associated */
+    AlertData alert_data = {
+        .subsystem = 1,
+        .componentId = 4,
+        .deviceId = 1,
+    };
+    AlertData *alert_data_cp = malloc(sizeof(AlertData));
+    *alert_data_cp = alert_data;
+    TEST_ASSERT_EQUAL(POST_DEV_CFG_DONE, LTC4015_fxnTable.cb_init(
+                                &gbc_pwr_int_bat_charger,
+                                &fact_lithiumIon_cfg, alert_data_cp));
+    TEST_ASSERT_EQUAL(16470, LTC4015_regs[0x01]);
```

Comment:
What does 16470 represent?  Those values should probably be tied to an enum, though I understand the usage predates this code change.

Reply:
I am working to create stub which will derive expected values using actual formulas.

Reply:
Code is modified as per the changes required.